src/Illuminate/Routing/Router.php

```diff
@@ -454,12 +462,39 @@ public function getLastGroupPrefix()
      * @param  string  $uri
      * @param  array|string|callable|null  $action
      * @return \Illuminate\Routing\Route
+     * @throws \Illuminate\Routing\Exceptions\RouteAlreadyRegisteredException
      */
     public function addRoute($methods, $uri, $action)
     {
+        $this->validateDuplicatesIfNeeded($methods, $uri);
+
         return $this->routes->add($this->createRoute($methods, $uri, $action));
     }
 
+    /**
+     * Throw an exception if a route with the same path already exists.
+     *
+     * @param  array|string  $methods
+     * @param  string  $uri
+     * @return void
+     * @throws \Illuminate\Routing\Exceptions\RouteAlreadyRegisteredException
+     */
+    protected function validateDuplicatesIfNeeded($methods, $uri)
+    {
+        if (! static::$enforceUniqueRoutes) {
+            return;
+        }
+
+        $cleanUriParameters = fn (string $uri) => preg_replace('/{(.*?)\}/s', '', $uri);
```

Comment:
This preg_replace doesn't seem right

Have a test with params like
```php
        $this->router->get('foo/{foo}/baz/{baz}', fn () => true);
        $this->router->get('foo/{bar}/bar/{baz}', fn () => false);
 ```
 
 Pretty sure it will throw
 
 You also need to account for the fact that params can be assigned different non overlapping "wheres" so you should replace them with their respecting where value instead of removing them

Reply:
Wheres are definitely tricky...

Reply:
Theoretically, you'll only know if the routes collide when a request comes in, right? So I guess it should not take them into account.